_blog/deploying-laravel-to-kubernetes/index.md

```diff
@@ -0,0 +1,294 @@
+
+So that you understand where you are at this moment, run the following command to get a list of desired and available Pods:
+
+```bash
+kubectl get deployment
+
+NAME                      DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+laravel-kubernetes-demo   1         1         1            1           57m
+```
+
+The output will be "1" for each. You want to have three available Pods so let's scale this up:
+
+```bash
+kubectl scale --replicas=3 deployment/laravel-kubernetes-demo
+deployment "laravel-kubernetes-demo" scaled
+```
+
+Done. You have replicated the first Pod to another two, giving you three Pods running this service. Running the `get deployment` command will verify this.
+
+```bash
+kubectl get deployment
+
+NAME                      DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+laravel-kubernetes-demo   3         3         3            3           59m
+```
+
+You can also see this in the Dashboard under Pods or in the Service detail screen.
+
+Now you're running three instances of the applications using three Pods.
+
+Imagine your application becoming even more popular. Thousands of visitors are using your website or software. In the past, you may have been busy writing more scripts to create more instances of your application. In Kubernetes you can scale to multiple instances in a snap:
+
+```bash
+kubectl scale --replicas=10 deployment/laravel-kubernetes-demo
+deployment "laravel-kubernetes-demo" scaled
+```
+
+You can see how convenient it is to use Kubernetes to scale your website.
+
+
+__Ingress__
+
+You've already achieved great things, you deployed the application and scaled the deployment. You have already seen the running application in the browser when pointed to the cluster's (Minikube) IP address and node's port number. Now, you will see how to access the application through an assigned URL as you would do when deploying to the cloud.
+
+To use a URL in Kubernetes, you need an Ingress. An Ingress is a set of rules to allow inbound connections to reach a Kubernetes cluster. The Ingress is necessary because, in Kubernetes, resources such as Pods only have IP addresses which are routable by and within the cluster. Meaning that they are not accessible or reachable to and from the world outside.
+
+I have included an `ingress.yaml` file with the source code of this demo application with the following contents:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: laravel-kubernetes-demo-ingress
+  annotations:
+    ingress.kubernetes.io/rewrite-target: /
+spec:
+  backend:
+    serviceName: default-http-server
+    servicePort: 80
+  rules:
+  - host: laravel-kubernetes.demo
+  - http:
```

Comment:
there's an error. should be `http` without the `-`. `host` & `http` belong to the same item.

Reply:
🤔 This works for me. Should I write it like this ?:

```yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: laravel-kubernetes-demo-ingress
  annotations:
    ingress.kubernetes.io/rewrite-target: /
spec:
  backend:
    serviceName: default-http-server
    servicePort: 80
  rules:
  - http:
      host: laravel-kubernetes.demo
      paths:
      - path: /
        backend:
          serviceName: laravel-kubernetes-demo
          servicePort: 8181
```

Reply:
I think so: https://kubernetes.io/docs/concepts/services-networking/ingress/#the-ingress-resource

Reply:
@danielepolencic I tried it and it doesn't work. The example link you've included does not have the _host_ key under http. The old way works.
![ingress_host_issue](https://user-images.githubusercontent.com/2071560/37809695-568cb08e-2e48-11e8-9ba2-11a10b135310.png)

Reply:
weird. Let's leave it like this then. sorry.

Reply:
I don't see it as weird because the host key is to set up a host on nginx (http) and not the actual Ingress. I assume we're still waiting for the images? You or I will also need to change the article date before merging.

Reply:
We have the image. It needs some tweaking. I also have the diagrams too. I'm a little bit swamped atm. As soon as I have more capacity I'll finish this off. Sorry for the delay.

Reply:
No worries. Once published I can also help with sharing the article 😄